eclipse-theia/theia
# Open VSX Registry integration

#### What it does
Widget where one can search and install VSCode Extensions from the Open VSX Registry.

<img width="1904" alt="Screenshot 2020-03-11 at 12 36 59" src="https://user-images.githubusercontent.com/3082655/76413165-3b09db00-6395-11ea-9d2a-fcae74aaab03.png">

TODO:
- [x] keyboard navigation does not work (rewrite lists with trees)
- [x] installed extensions metadata should not be fetched from the registry, but should use local
- [x] vscode extension URI resolver should be installed, see https://github.com/eclipse-theia/theia/pull/6655#issuecomment-593582431
- [x] extensions should be installed under `${configDir}/extensions` by default
- [x] extensions should be loaded from `${configDir}/extensions` by default
- [x] focus is not properly given on the view activation
- [x] support built-in extensions
- [x] update view container layout based on query, preserve layout state for different modes
- [x] progress reporting in view container for search, install, uninstall
- [x] theming
- [x] preserve state properly, i.e. search query should be preserved only, everything else should be computed from it again
- [x] check TODOs in code

Out of scope:
- finding a version compatible extension
- detecting outdated extensions and upgrading them
- supporting workspace extensions (only built-in and global (user))
- supporting extension enablement
- use webview to render remote content like README

#### How to test

- Try to install/uninstall extensions. Make sure that they are preserved (or not after uninstall) on restart.
- Look at the extension editor.
- Test that the extension editor is opened for the vscode extension URIs.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
